KS-2-3 Train

CasTeTRU opened the pull request.

CasTeTRU committed:
KS-2-3 Train

CasTeTRU committed:
KS-2-3 Train
Convert minutes to hours and output time of arrival

ievetrovdev commented on src/main/kotlin/lesson_2/lesson2_task3.kt:
```diff
@@ -0,0 +1,10 @@
+    val travelTime: Short = 457
+    val arrivalHour: Int = (departureHour * 60 + departureMinute + travelTime) / 60
+    val arrivalMinute: Int = (departureHour * 60 + departureMinute + travelTime) % 60
+    println("$arrivalHour:$arrivalMinute")
```
А вот здесь как раз форматирование важно! Время хоть и выводится в формате [час:минуты], но не форматируется для двузначного представления минут и часов, попробуй посмотреть в сторону String.format

ievetrovdev replied:
По поводу разделения кода на блоки, отписал в комментах к прошлым задачам

ievetrovdev commented on src/main/kotlin/lesson_2/lesson2_task3.kt:
```diff
@@ -0,0 +1,10 @@
+    val departureHour: Short = 9
+    val departureMinute: Short = 39
+    val travelTime: Short = 457
+    val arrivalHour: Int = (departureHour * 60 + departureMinute + travelTime) / 60
```
тобы снизить риск ошибок, предлагаю еще вынести 60 в константу. Кроме того это [магическое число](https://ru.wikipedia.org/wiki/%D0%9C%D0%B0%D0%B3%D0%B8%D1%87%D0%B5%D1%81%D0%BA%D0%BE%D0%B5_%D1%87%D0%B8%D1%81%D0%BB%D0%BE_(%D0%BF%D1%80%D0%BE%D0%B3%D1%80%D0%B0%D0%BC%D0%BC%D0%B8%D1%80%D0%BE%D0%B2%D0%B0%D0%BD%D0%B8%D0%B5)#%D0%9F%D0%BB%D0%BE%D1%85%D0%B0%D1%8F_%D0%BF%D1%80%D0%B0%D0%BA%D1%82%D0%B8%D0%BA%D0%B0_%D0%BF%D1%80%D0%BE%D0%B3%D1%80%D0%B0%D0%BC%D0%BC%D0%B8%D1%80%D0%BE%D0%B2%D0%B0%D0%BD%D0%B8%D1%8F), что тоже не является хорошей практикой.

ievetrovdev reviewed (changes requested):
А также обрати внимание на нейминг ветки и ПР

CasTeTRU committed:
KS-2-3-Train
Вместо магического числа добавлена константа времени
Сделано форматирование по абзацам
Вывод в консоль переделан под отображение представления минут и часов

CasTeTRU commented:
> А также обрати внимание на нейминг ветки и ПР

Вот здесь немного не понимаю:
Нейминг ветки: "_KS-2-3-Train_" - корректно? Номер урока + краткое описание, что решаем в задаче
Нейминг ПР "_KS-2-3-Train
Вместо магического числа добавлена константа времени
Сделано форматирование по абзацам
Вывод в консоль переделан под отображение представления минут и часов_" - Повторяю название ветки + расписываю что было изменено.
Или тут другая логика?

ievetrovdev commented:
> Нейминг ветки: "_KS-2-3-Train_" - корректно? Номер урока + краткое описание, что решаем в задаче
> Нейминг ПР "_KS-2-3-Train_

В целом это не так критично. Но на больших проектах, где задействовано много разработчиков, корректнее писать какое действие описывается в коде, чтобы человеку, первый раз увидевшему код, было легко ориентироваться. В данном случае, например так - KS-2-3-Train-arrival-calculation
